Modules/Devices/src/Devices.Application/Devices/Queries/GetActiveDevice/Handler.cs

```diff
@@ -15,24 +15,20 @@ public class Handler : IRequestHandler<GetActiveDeviceQuery, DeviceDTO>
     private readonly IDevicesDbContext _dbContext;
     private readonly IMapper _mapper;
     private readonly IUserContext _userContext;
+    private readonly IDevicesRepository _devicesRepository;
 
-    public Handler(IDevicesDbContext dbContext, IMapper mapper, IUserContext userContext)
+    public Handler(IDevicesDbContext dbContext, IMapper mapper, IUserContext userContext, IDevicesRepository devicesRepository)
     {
         _dbContext = dbContext;
         _mapper = mapper;
         _userContext = userContext;
+        _devicesRepository = devicesRepository;
     }
 
     public async Task<DeviceDTO> Handle(GetActiveDeviceQuery request, CancellationToken cancellationToken)
     {
-        var device = await _dbContext
-            .SetReadOnly<Device>()
-            .NotDeleted()
-            .WithId(_userContext.GetDeviceId())
-            .IncludeUser()
-            .ProjectTo<DeviceDTO>(_mapper.ConfigurationProvider)
-            .FirstOrDefaultAsync(cancellationToken);
-
-        return device;
+        var device = await _devicesRepository.GetCurrentDevice(_userContext.GetDeviceId(), cancellationToken);
```

Comment:
Use the `GetDeviceById` method instead. The repository doesn't know anything about "current" devices/identities